cmd/sonarExecuteScan.go

```diff
@@ -411,3 +415,31 @@ func getTempDir() string {
 	}
 	return tmpFolder
 }
+
+// Fetches parameters from environment variables and updates the options accordingly (only if not already set)
+func detectParametersFromCI(options *sonarExecuteScanOptions) {
+	provider, err := orchestrator.NewOrchestratorSpecificConfigProvider()
+	if err != nil {
+		log.Entry().WithError(err).Warning("Cannot infer config from CI environment")
+		return
+	}
+
+	if provider.IsPullRequest() {
+		config := provider.GetPullRequestConfig()
+		if len(options.ChangeBranch) == 0 {
+			options.ChangeBranch = config.Branch
+		}
+		if len(options.ChangeTarget) == 0 {
+			options.ChangeTarget = config.Base
+		}
+		if len(options.ChangeID) == 0 {
+			options.ChangeID = config.Key
+		}
+	} else {
+		config := provider.GetBranchBuildConfig()
+
+		if len(options.BranchName) == 0 {
+			options.BranchName = config.Branch
+		}
```

Comment:
In a Sonar context this could be an issue as the branch is detected as *long living branch* and not as main branch. In general this parameter is not used.
I guess in a CI environment the `branchName` is always set. Maybe we need to have another parameter `inferBranchName` to be compatible.

Reply:
✅